client/client.go

```diff
@@ -363,6 +364,28 @@ func (c *Client) init() error {
 	return nil
 }
 
+// ReloadTLSConnectoins allows a client to reload RPC connections if the
+// client's TLS configuration changes from plaintext to TLS
+func (c *Client) ReloadTLSConnections(newConfig *nconfig.TLSConfig) error {
+	c.configLock.Lock()
+	c.config.TLSConfig = newConfig
+	c.configLock.Unlock()
+
+	if c.config.TLSConfig.EnableRPC {
```

Comment:
Why is this only checking for enableRPC and not for enableHttp?

Reply:
HTTP connections are handled at the HTTP server level